packages/webpack/externals-loading-webpack-plugin/src/index.ts

```diff
@@ -0,0 +1,359 @@
+// Copyright 2025 The Lynx Authors. All rights reserved.
+// Licensed under the Apache License Version 2.0 that can be found in the
+// LICENSE file in the root directory of this source tree.
+
+/**
+ * @packageDocumentation
+ *
+ * A webpack plugin to load externals in Lynx. Use `lynx.fetchBundle()` and `lynx.loadScript()` API to load and parse the externals.
+ *
+ * @remark
+ * It requires Lynx >= 3.4.
+ */
+
+import type {
+  Compiler,
+  ExternalItemFunctionData,
+  ExternalItemValue,
+  ExternalsType,
+} from '@rspack/core';
+
+/**
+ * The options of the `ExternalsLoadingPlugin`.
+ *
+ * @public
+ */
+export interface ExternalsLoadingPluginOptions {
+  /**
+   * The chunk names to be considered as main thread chunks.
+   */
+  mainThreadChunks: string[];
+
+  /**
+   * The chunk names to be considered as background chunks.
+   */
+  backgroundChunks: string[];
+
+  /**
+   * The name of the main thread layer.
+   */
+  mainThreadLayer: string;
+
+  /**
+   * The name of the background layer.
+   */
+  backgroundLayer: string;
+
+  /**
+   * Specify the externals to be loaded. The externals should be Lynx Bundles.
+   *
+   * @example
+   *
+   * Load `lodash` library in background layer and `main-thread` layer.
+   *
+   * ```js
+   * module.exports = {
+   *  plugins: [
+   *    new ExternalsLoadingPlugin({
+   *      externals: {
+   *        lodash: {
+   *          libraryName: "Lodash",
```

Comment:
I think we can delete this config since it is only used by us internally. We can generate a unique id for each external module.